command/var_init.go

```diff
@@ -0,0 +1,212 @@
+		return 1
+	}
+
+	// Write out the example
+	err = ioutil.WriteFile(fileName, []byte(fileContent), 0660)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Failed to write %q: %v", fileName, err))
+		return 1
+	}
+
+	// Success
+	if !quiet {
+		c.Ui.Warn(WrapAndPrepend(TidyRawString(msgWarnKeys), 70, ""))
+		c.Ui.Output(fmt.Sprintf("Example secure variable specification written to %s", fileName))
+	}
+	return 0
+}
+
+const (
+	msgWarnKeys = `
+	REMINDER: While keys in the 'Items' collection can contain dots, using
+	them in templates is easier when they do not. As a best practice, avoid
+	dotted keys when possible.`
+)
+
+var defaultHclVarSpec = strings.TrimSpace(`
+# A secure variable Path can be specified in the specification file
+# and will be used when writing the variable without specifying a
+# Path in the command or when writing JSON directly to the `+"`/var/`"+`
+# HTTP API endpoint
+Path = "path/to/variable" 
+
+# The Namespace to write the variable can be included in the specification
+# and is the highest precedence way to set the namespace value.
+Namespace = "default"
+
+# The Items collection is the only strictly required part of a secure
+# variable specification. It contains the sensitive material to encrypt
+# and store as a Nomad secure variable. The entire Items collection are
+# encrypted and decrypted as a single unit.
+
+`+warnInHCLFile()+`
+Items {
+  key1 = "value 1"
+  key2 = "value 2"
+}
+`) + "\n"
+
+var defaultJsonVarSpec = strings.TrimSpace(`
+{
+  "Items": {
+    "key1": "value 1",
+    "key2": "value 2"
+  }
+}
+`) + "\n"
+
+func warnInHCLFile() string {
+	return WrapAndPrepend(TidyRawString(msgWarnKeys), 70, "# ")
+}
+
+// WrapString is a convienience func to abstract away the word wrapping
+// implementation
+func WrapString(input string, lineLen int) string {
+	return wordwrap.String(input, lineLen)
+}
+
+// WrapAndPrepend will word wrap the input string to lineLen characters and
+// prepend the provided prefix to every line. The total length of each returned
+// line will be at most len(input[line])+len(prefix)
+func WrapAndPrepend(input string, lineLen int, prefix string) string {
+	ss := strings.Split(wordwrap.String(input, lineLen), "\n")
+	prefixStringList(ss, prefix)
+	return strings.Join(ss, "\n")
+}
+
+// HangingIndent will indent all lines except the first line by the
+// given indent value, wrapping all lines at lineLen.
+func HangingIndent(input string, lineLen int, indent int) string {
+	ss := strings.Split(wordwrap.String(input, lineLen), "\n")
+	// wrap as though indent was not in play at all and get the first line
+	if len(ss) == 1 {
+		return ss[0]
+	}
+	var out strings.Builder
+	out.WriteString(ss[0])
+
+	// reassemble and split the wraps taking the indent into account
+	input = strings.Join(ss[1:], "\n")
+	ss = strings.Split(wordwrap.String(input, lineLen-indent), "\n")
+	prefixStringList(ss, strings.Repeat(" ", indent))
+
+	return strings.Join(ss, "\n")
+}
```

Comment:
This function is unused. Is this to support an incoming PR? (In which case, that's fine to leave)

Reply:
Pulling it for now.